src/blockchain/log-processors/profit-loss/frozen-funds.ts

```diff
@@ -0,0 +1,99 @@
+}
+
+interface FrozenFundsParams {
+  frozenFundsBeforeEvent: FrozenFunds; // FrozenFunds prior to processing this FrozenFundsEvent
+  event: FrozenFundsEvent; // FrozenFundsEvent to process and return updated FrozenFunds as impacted by this event
+}
+
+// TODO doc - frozen funds in context of one (market, outcome)
+// TODO unit test
+export function getFrozenFundsAfterEventForOneOutcome(params: FrozenFundsParams): FrozenFunds {
+  if (params.event === "ClaimProceeds") {
+    return {
+      // TODO doc
+      frozenFunds: ZERO,
+      frozenProfit: ZERO,
+    };
+  }
+  // params.event is a trade executed; TODO break each event into a separate function
+  const trade = params.event.tradeData;
+  const frozenFundsAfterEvent = Object.assign({}, params.frozenFundsBeforeEvent);
+
+  // TODO make this into a helper
+  /*
+  function getTokensForDestroyedSharesForTrade(): {
+    creatorTokens: BigNumber,
+    fillerTokens: BigNumber,
+  } {
+    return {
+      creatorTokens: ZERO,
+      fillerTokens: ZERO,
+    };
+  }
+  */
+  const tokensFreedFromEscrowByThisTrade = BigNumber.min(trade.numCreatorShares, trade.numFillerShares).multipliedBy(ZERO /* TODO displayRange */);
+
+  // TODO Alex - can you confirm formula for this:
+  const creatorPortionOfTokensFreedFromEscrow = tokensFreedFromEscrowByThisTrade; // TODO actual value I think is f(minPrice, maxPrice, price)?
+  const fillerPortionOfTokensFreedFromEscrow = tokensFreedFromEscrowByThisTrade; // TODO actual value I think is f(minPrice, maxPrice, price)?
+
+  if (params.event.creatorOrFiller === "creator") {
+    // I created this trade
+    // TODO doc formulas
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.minus(creatorPortionOfTokensFreedFromEscrow);
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.plus(trade.numCreatorTokens);
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.minus(trade.numFillerTokens);
+  } else {
+    // I filled this trade
+    // TODO doc formulas
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.minus(fillerPortionOfTokensFreedFromEscrow);
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.plus(trade.numFillerTokens);
+    frozenFundsAfterEvent.frozenFunds = frozenFundsAfterEvent.frozenFunds.minus(trade.numCreatorTokens);
+  }
+
+  if (frozenFundsAfterEvent.frozenFunds.isGreaterThanOrEqualTo(
+    params.frozenFundsBeforeEvent.frozenFunds)) {
+    // frozen funds increased or stayed the same, so my available funds decreased or
+    // stayed the same, which means that my frozen profit should increase by realized
+    // profit and loss for this trade. Ie. frozen profit is profit that I could claim,
+    // but haven't, and we detect this by looking for a non-increase in available funds.
+    // TODO Alex - is this right?
+    frozenFundsAfterEvent.frozenProfit = frozenFundsAfterEvent.frozenProfit.plus(params.event.realizedProfit);
```

Comment:
frozen profit doesn't need to be stored outside of frozen funds, its just a part of calculating the frozen funds and wont be displayed or used in its own right. This should just be `frozenFunds += params.event.realizedProfit`

frozen profit is profit or loss which is not available in the users funds as it only applied in the context of adjusting the entry into a new position. It's not profit that you could claim but haven't, that's _unrealized_ profit.